mllib-local/src/test/scala/org/apache/spark/ml/linalg/VectorsSuite.scala

```diff
@@ -72,6 +72,12 @@ class VectorsSuite extends SparkMLFunSuite {
     }
   }
 
+  test("sparse vector construction with negative indices") {
+    intercept[IllegalArgumentException] {
+      Vectors.sparse(3, Array(1, -2), Array(3.0, 5.0))
```

Comment:
This actually won't test the check for negative indices, because the indices are missorted.